packages/garbo/src/tasks/freeGiantSandworm.ts

```diff
@@ -331,10 +343,17 @@ function drumMachineWorthIt(): boolean {
     const rate =
       REJECTION *
       clamp(BASE_RATE * (1 + (getModifier("Item Drop") * squint) / 100), 0, 1);
-    _drumMachineWorthIt =
-      mallPrice($item`drum machine`) < rate * garboValue($item`spice melange`);
+    _drumMachineROI =
+      rate * garboValue($item`spice melange`) - mallPrice($item`drum machine`);
   }
-  return _drumMachineWorthIt;
+  return _drumMachineROI;
+}
+
+function drumMachineWorthIt(): boolean {
+  if (_drumMachineWorthIt === undefined) {
+    _drumMachineWorthIt = drumMachineROI() > 0;
+    return _drumMachineWorthIt;
+  } else return _drumMachineWorthIt;
```

Comment:
```suggestion
function drumMachineWorthIt(): boolean {
  return drumMachineROI() > 0;
```
no need to cache this, it's not difficult at all